MercurialExtensions/fixutf8/fixutf8.py

```diff
@@ -185,41 +143,65 @@ def localize(h):
 			getbuffers = lambda ui: ui._buffers
 		else:
 			getbuffers = lambda ui: ui.buffers
+
 		def f(orig, ui, *args, **kwds):
 			if not getbuffers(ui):
 				win32helper.rawprint(h, ''.join(args))
 			else:
 				orig(ui, *args, **kwds)
+
 		return f
 
 	extensions.wrapfunction(_ui.ui, "write", localize(win32helper.hStdOut))
 	extensions.wrapfunction(_ui.ui, "write_err", localize(win32helper.hStdErr))
 
-def extsetup(ui):
-	#print "extsetup start"
+	'''
+	The following function has nothing to do with utf8 and could potentially belong
+	in a separate extension. The old versions of mercurial allowed branch names that
+	were only integers. Starting in version 3 integer branch names were disallowed.
+	Chorus uses branch names to handle model upgrades. This change would force us to
+	simulate a model upgrade with Chorus clients that used only the model version number
+	as a branch name. Alternatively we can wrap the function and skip the integer name
+	test.
+	'''
```

Comment:
As far as I can tell they just decided that it was a bad idea. They do not force old repositories to change number only branch names as that would break their backward compatibility. All of their code must still work with them. We could certainly avoid use of number only branch names in the future just in case they do something more drastic in Mercurial 4, but that change would need to be made in the clients.